syntax = "proto3";

package common;

option python_generic_services = true;

// HealthCheckRequest is the request for health check
message HealthCheckRequest {
  // Service name to check (empty for overall health)
  string service = 1;
}

// HealthCheckResponse is the response for health check
message HealthCheckResponse {
  // ServingStatus represents the health status
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;
  }
  
  ServingStatus status = 1;
}

// Health service definition (compatible with grpc-health-checking)
service Health {
  // Check returns the serving status of the service
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
  
  // Watch streams health status updates
  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}
